pkg/ipam/node.go

```diff
@@ -219,34 +219,28 @@ func (n *Node) loggerLocked() (logger *logrus.Entry) {
 func (n *Node) getMaxAboveWatermark() int {
 	if n.resource.Spec.IPAM.MaxAboveWatermark != 0 {
 		return n.resource.Spec.IPAM.MaxAboveWatermark
-	}
-	// OBSOLETE: This can be removed in Cilium 1.9
-	return n.resource.Spec.ENI.MaxAboveWatermark
+	}	
+	return nil
 }
 
 // getPreAllocate returns the pre-allocation setting for an AWS node
 //
 // n.mutex must be held when calling this function
-func (n *Node) getPreAllocate() int {
+func (n *Node) getPreAllocate() int {	
 	if n.resource.Spec.IPAM.PreAllocate != 0 {
 		return n.resource.Spec.IPAM.PreAllocate
-	}
-	// OBSOLETE: This can be removed in Cilium 1.9
-	if n.resource.Spec.ENI.PreAllocate != 0 {
-		return n.resource.Spec.ENI.PreAllocate
-	}
+	}	
 	return defaults.IPAMPreAllocation
 }
 
 // getMinAllocate returns the minimum-allocation setting of an AWS node
 //
 // n.mutex must be held when calling this function
-func (n *Node) getMinAllocate() int {
+func (n *Node) getMinAllocate() int {	
 	if n.resource.Spec.IPAM.MinAllocate != 0 {
 		return n.resource.Spec.IPAM.MinAllocate
-	}
-	// OBSOLETE: This can be removed in Cilium 1.9
-	return n.resource.Spec.ENI.MinAllocate
+	}	
+	return nil
```

Comment:
Same as above, `nil` cannot be returned here. The if statement is also not necessary anymore.